tests/amr/data/particles/schedule/test_tag_strategy.h

```diff
@@ -79,29 +78,44 @@ class TagStrategy : public SAMRAI::mesh::StandardTagAndInitStrategy
 
                     auto particlesBox = particlesData->getBox();
 
-                    // here we are 1D
-                    for (int iCellPos = particlesBox.lower(dirX);
-                         iCellPos <= particlesBox.upper(dirX); ++iCellPos)
+                    auto lowerZ = dimension == 3 ? particlesBox.lower(dirZ) : 0;
+                    auto upperZ = dimension == 3 ? particlesBox.upper(dirZ) : 0;
+
+                    auto lowerY = dimension == 2 ? particlesBox.lower(dirY) : 0;
+                    auto upperY = dimension == 2 ? particlesBox.upper(dirY) : 0;
+
+                    auto lowerX = particlesBox.lower(dirX);
+                    auto upperX = particlesBox.upper(dirX);
+
+                    for (auto iCellZ = lowerZ; iCellZ <= upperZ; ++iCellZ)
                     {
-                        float middle = 0.5;
-                        float delta  = 0.30;
+                        for (auto iCellY = lowerY; iCellY <= upperY; ++iCellY)
+                        {
+                            for (auto iCellX = lowerX; iCellX <= upperX; ++iCellX)
+                            {
+                                std::array<int, 3> _3DiCell = {iCellX, iCellY, iCellZ};
+                                float middle                = 0.5;
+                                float delta                 = 0.30;
 
-                        Particle<dimension> particle;
+                                Particle<dimension> particle;
 
-                        particle.weight = 1.;
-                        particle.charge = 1.;
-                        particle.v      = {{1.0, 0.0, 0.0}};
+                                for (size_t i = 0; i < dimension; i++)
+                                    particle.iCell[i] = _3DiCell[i];
 
-                        particle.iCell[dirX] = iCellPos;
+                                particle.delta.fill(middle);
+                                particle.weight = 1.;
+                                particle.charge = 1.;
+                                particle.v      = {{1.0, 0.0, 0.0}};
 
-                        particle.delta[dirX] = middle - delta;
-                        interior.push_back(particle);
+                                interior.push_back(particle);
 
-                        particle.delta[dirX] = middle + delta;
-                        interior.push_back(particle);
+                                particle.delta[dirX] = middle - delta;
```

Comment:
cannot be always dirX

Reply:
can you elaborate on this please

Reply:
I don't know to why extent the dispatching matters for the tests (I don't remember) but I find it weird if it's 2D to put 3 particles in the cell in the same way it is done in 1D. I'd had imagined something more 2D that dispatch them in the cell with a delta in Y too.

Reply:
For what the test does I'm not sure it matters.

consider the test names.

```
ghostParticleNumberisZeroBeforeScheduleAndCorrectAfterSchedule;
hasGhostParticlesInGhostAMRCellsAfterSchedule;
hasGhostParticleFilledFromNeighborFirstCell;
```